README.md

```diff
@@ -177,69 +176,68 @@ you need to use `IComponentVisitor`.
   ```
 ## Sheets API
 
-The base unit of sheets API is `ISheetSegment` interface, it provides an interface to get components for different
-sheet section parts (header, rows, footer).
-
-```csharp
-public interface ISheetSegment<THeaderData, TRowData, TFooterData>
-{
-    IEnumerable<IComponent> BuildHeaders(THeaderData data);
-
-    IEnumerable<IComponent> BuildRows(HeaderRowData<THeaderData, TRowData> data, int rowIndex);
-    
-    IEnumerable<IComponent> BuildFooters(HeaderFooterData<THeaderData, TFooterData> data);
-}
-```
-
 There are two main kinds of sheet segments:
 
-- Plain `SheetSegmentBase`\
-  A segment that consists of a single data column.
-- A `PrototypeSheetSegmentBase`\
-  A segment that can represent a collection of data columns.\
+- A segment that consists of a single data column:
+  - `SegmentBase`
+  - `HeaderRowSegmentBase`
+- A segment that can represent a collection of data columns:
+  - `PrototypeSegmentBase`
+  - `PrototypeHeaderRowSegmentBase`
+  
 
-### SheetSegmentBase
+### SegmentBase
 
 - Requires you to implement 2 methods
     - `IComponent BuildHeader(THeaderData data)`
     - `IComponent BuildRow(HeaderRowData<THeaderData, TRowData> data, int rowIndex)`
-- With an option to overload
     - `IComponent BuildFooter(HeaderFooterData<THeaderData, TFooterData> data)`
 
-### PrototypeSheetSegmentBase
+### HeaderRowSegmentBase
 
-Prototype sheet segments are useful when you table may have a dynamic amount of columns depending on the data that has
+- Requires you to implement 2 methods
+    - `IComponent BuildHeader(THeaderData data)`
+    - `IComponent BuildRow(HeaderRowData<THeaderData, TRowData> data, int rowIndex)`
+
+### PrototypeSegmentBase
+
+Prototype sheet segments are useful when your table may have a dynamic amount of columns depending on the data that has
 been provided.
 
-To be a part of a sheet segment collection, prototype must implement `ISheetSegment<THeaderData, TRowData, TFooterData>`
-interface,
-just like any other static sheet segment. But prototype segment must have a collection of header data to create multiple
-segments, so
-you must implement `IEnumerable<TPrototypeHeaderData> SelectHeaderData(THeaderData data)` method to extract collection
-of header data
-that prototypes will use from general header data.
+Prototype segment must have a collection of header data to create multiple segments, so
+you must implement `IEnumerable<TSourceHeaderData> SelectHeaderData(THeaderData data)` method to extract collection
```

Comment:
The method called just `Select`, but it would be better off called `SelectHeaderData`.